### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by adrianreber**
README.md contained a lot of information which is no longer correct. This updates README.md to have up to date content and also adds a markdown linter.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,60 +1,64 @@
...
-The Kubernetes kubelet creates checkpoints which consists of metadata (`checkpointed.pods`)
-and tar archives containing the actual pod checkpoints.
+Container engines like *Podman* and *CRI-O* have the ability to checkpoint a
+container.  All data related to a checkpoint is collected in a checkpoint
+archive. With the help of this tool, `checkpointctl`, it is possible to display
+information about these checkpoint archives.
 
-With the help of this tool, `checkpointctl`, it is possible to display, extract or insert
-checkpoints.
+Details how to create checkpoints with the help of [CRIU][criu] can be found:
```
Comment (rst0git) [inline README.md:11]:
```Suggestion
Details on how to create checkpoints with the help of [CRIU][criu] can be found at:
```

File: .github/workflows/markdown-lint.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+name: markdown-lint
+
+on: [push, pull_request]
```
Comment (rst0git) [inline .github/workflows/markdown-lint.yml:3]: Should we add [`paths`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-including-paths) to run this workflow only when `README.md` has been modified?

Comment (adrianreber) [reply .github/workflows/markdown-lint.yml:3]: done

**Review by rst0git — APPROVED** [review]
LGTM

**Pull request merged**